python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/_utils.py

```diff
@@ -0,0 +1,134 @@
+import json
+import logging
+import warnings
+from enum import Enum
+from importlib.metadata import version
+from typing import (
+    Any,
+    Iterator,
+    List,
+    Mapping,
+    NamedTuple,
+    Optional,
+    Protocol,
+    Sequence,
+    Tuple,
+    Union,
+    cast,
+)
+
+from openinference.instrumentation.openai._with_span import _WithSpan
+from openinference.semconv.trace import SpanAttributes
+from opentelemetry import trace as trace_api
+from opentelemetry.util.types import Attributes, AttributeValue
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+_OPENAI_VERSION = tuple(map(int, version("openai").split(".")[:3]))
```

Comment:
nit: if the module itself is also `_` prefixed, can't we largely avoid a lot of this `_` prefixing? I get that this is the pythonic way but It feels largely just difficult to read for me. No need to respond on this but I personally find it kinda silly TBH. I kinda would prioritize readability over pseudo privatization.

Reply:
I know, haha, but the thing is that it's not really "pseudo" any more. If you look at [this](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/14d7779366ec99800f1e9a1f7137c01bc1aff617/opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py#L25) for example, the underscore is in fact a real signaling mechanism.

Reply:
I see... I'll get used to it :bleeding eyes: